File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    static bool comp(vector<int>& a, vector<int>& b) {
        return a[0]<b[0];
    }
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end(),comp);
        int start = intervals[0][0], end = intervals[0][1];
        vector<vector<int>> ans;
        for(int i=1;i<intervals.size();i++) {
            if(end<intervals[i][0] ) {
                ans.push_back({start,end});
                start=intervals[i][0];
                end=intervals[i][1];
            } else {
                start = min(start,intervals[i][0]);
                end = max(end,intervals[i][1]);
            }
        }
        ans.push_back({start,end});
        return ans;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals,new Comparator<int[]>() {
            @Override
            public int compare(int[] a, int[] b) {
                return Integer.compare(a[0],b[0]);
            }
        });
        List<List<Integer>> arr = new ArrayList<>();
        int start = intervals[0][0], end = intervals[0][1];
        for(int i=1;i<intervals.length;i++) {
            if(intervals[i][0]>end) {
                arr.add(List.of(start,end));
                start = intervals[i][0];
                end = intervals[i][1];
            } else {
                start = Math.min(start,intervals[i][0]);
                end = Math.max(end,intervals[i][1]);
            }
        }
        arr.add(List.of(start,end));
        int[][] ans = new int[arr.size()][2];
        for(int i=0;i<arr.size();i++) {
            ans[i][0] = arr.get(i).get(0);
            ans[i][1] = arr.get(i).get(1);
        }
        return ans;
    }
}